Reflection/Login.simba

```diff
@@ -0,0 +1,299 @@
+    R_OpenGametab(GAMETAB_LOGOUT);
+  Wait(50, RandomRange(250, 692));
+
+  if (RSWidget.IsValid(R_WORLD_SWITCHER_CONTAINER.Group, R_WORLD_SWITCHER_CONTAINER.Child)) and (not R_WidgetIsHidden(R_WORLD_SWITCHER_CLOSE)) then
+  begin
+    Widget := RSWidget.Get(R_WORLD_SWITCHER_CLOSE.Group, R_WORLD_SWITCHER_CLOSE.Child);
+    if Widget.ref <> nil then
+    begin
+      Mouse.Click(Widget.Bounds, MOUSE_LEFT);
+      Result := True;
+      Widget.Free;
+    end;
+  end
+
+end;
+
+{*
+R_OpenWorldSwitcher
+~~~~~~~~~~~~~~~~~~~~~~
+Opens the world switcher.
+
+Example:
+    R_OpenWorldSwitcher;
+*}
+Function R_OpenWorldSwitcher: Boolean;
+var
+  Widget: RSWidget;
+begin
+  if (not R_IsGametabOpen(GAMETAB_LOGOUT)) then
+    R_OpenGametab(GAMETAB_LOGOUT);
+
+  Wait(50, RandomRange(250, 692));
+
+  if (not R_WaitUntilVisible(R_LOGOUT_CONTAINER, 2500)) or R_WidgetIsHidden(R_LOGOUT_WORLD_SWITCHER_BUTTON) then
+    Exit(False);
+
+  Widget := RSWidget.Get(R_LOGOUT_WORLD_SWITCHER_BUTTON.Group, R_LOGOUT_WORLD_SWITCHER_BUTTON.Child);
+  if Widget.ref <> nil then
+  begin
+    Mouse.Click(Widget.Bounds, MOUSE_LEFT);
+    Result := R_WaitUntilVisible(R_WORLD_SWITCHER_SCROLL_UP, 2500);
+    Widget.Free;
+  end;
+end;
+
+{*
+R_WorldToBox
+~~~~~~~~~~~~~~~~~~~~~~
+Utility function that converts a given world to mainscreen coordinates.
+
+Example:
+    R_WorldToBox(500);
+*}
+Function R_WorldToBox(World: Int32): TBox;
+var
+  WorldWidget: RSWidget;
+begin
+  WorldWidget := RSWidget.Get(R_WORLD_SWITCHER_LIST.Group, R_WORLD_SWITCHER_LIST.Child, World);
+  Result := WorldWidget.Bounds;
+  WorldWidget.Free;
+end;
+
+{*
+R_ScrollToWorld
+~~~~~~~~~~~~~~~~~~~~~~
+Scrolls to the proper world in the world switcher.
+
+Example:
+    R_ScrollToWorld(500, TRSWorldScroll.SCROLLBAR);
+*}
+Function R_ScrollToWorld(World: Int32; ScrollType: TRSWorldScroll = TRSWorldScroll.SCROLLWHEEL): Boolean;
+var
+  ScrollBar, ScrollArrow, WorldContainer: RSWidget;
+  WorldContainerBounds: TBox;
+  T: Timer;
+  Direction: Boolean;
+  P: TPoint;
+begin
+  WorldContainer := RSWidget.Get(R_WORLD_SWITCHER_LIST_INTERFACE_CONTAINER_CHILD.Group, R_WORLD_SWITCHER_LIST_INTERFACE_CONTAINER_CHILD.Child);
+  if WorldContainer.ref = nil then
+    Exit(False);
+
+  //Adds a buffer to make sure that the world is completely clickable
+  WorldContainerBounds := [WorldContainer.Bounds.X1, WorldContainer.Bounds.Y1 + 15, WorldContainer.Bounds.X2, WorldContainer.Bounds.Y2 - 15];
+  WorldContainer.Free;
+
+  if WorldContainerBounds.Contains(R_WorldToBox(World)) then
+    Exit(True);
+
+  Direction := WorldContainerBounds.Middle.Y > R_WorldToBox(World).Middle.Y;
+
+  //Scroll the interface using the mouse-wheel
+  if ScrollType = TRSWorldScroll.SCROLLWHEEL then
+  begin
+    Mouse.Move(WorldContainerBounds);
+
+    T.Start;
+    while not WorldContainerBounds.Contains(R_WorldToBox(World)) do
+    begin
+      if T.ElapsedTime > 5000 then
+        Exit(False);
+
+      Mouse.Scroll(3, Direction);
```

Comment:
Maybe
```suggestion
      Mouse.Scroll(Random(1, 3), Direction);
```
Would be better. I haven't tested to see how far 1 scroll actually scrolls though, Maybe even `Random(2, 6)` would be better if its not too much.

Reply:
I know that the increment of the Mouse scroll is normally constant for a specific device unless changed in the device's settings. I would assume that having it randomly change might be a possible red flag?

Reply:
yeah good point. I'm fine with leaving it as 3